### PR TITLE
Milestone 1: OOP School Library:  Add Person, Student and Teacher Classes

### PR DESCRIPTION
**Pull request by Miliyonayalew**
## In this Milestone I have
- Initialized Repo
- Installed `Rubocop`
- Implemented classes and objects in Ruby.
- Implemented encapsulation and inheritance with Ruby.
- Created `Person` class as the project requirment.
- Created `Student` and `Teacher` classes by Inheriting from the `Person` class as the project requirment.

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+class Person
+  def initialize(age, name: 'Unknown', parent_permission: true)
+    @id = Random.rand(1..1000)
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+  end
+
+  attr_reader :id
+  attr_accessor :@name, :@age
```
Comment (zieeco) [inline person.rb]: - This is how to write the getters and setters `attr_accessor :name, :age` let's fix this, and to improve readability, let's move it up before the constructor

**Review by zieeco — CHANGES_REQUESTED** [review]
Hi @Miliyonayalew ,


Great job so far! 👏 
You've done well, however, there are some issues that you still need to work on to go to the next project, but you are almost there!


## To highlight 🎯

- No linter errors.✔️
- Nice classes. ✔️
- Descriptive PR. ✔️


## Required Changes ♻️
_Check the comments under the review._





## Optional suggestions
- N/A


**_Every comment with the **[OPTIONAL]** prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._**


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb (hunk 1 above)
Comment (Miliyonayalew) [reply person.rb]: @zieeco Thanks for the review and your valuable feedback. changes are made, Happy Coding :smile:

**Review by FrederickMih — APPROVED** [review]
## Hi :raising_hand:  @Miliyonayalew, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- getters and setters correctly defined:+1: 
- All Linter checks are passing :heavy_check_mark: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**